### ISSUE TITLE
refactor: migrate to rule-map config system and bump major version to v2

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#82]
## Background

As gomarklint's rule set grows (tracked in #76), the current configuration design does not scale well. Each new rule requires:

- A new `bool` field in `Config` struct (e.g., `EnableXxxCheck bool`)
- A new CLI flag (e.g., `--enable-xxx-check`)
- A new conditional block in `linter.go`

With 20+ candidate rules on the roadmap, this approach becomes unmaintainable.

A comparison with the major Markdown linter ecosystem (markdownlint, remark-lint) also reveals two additional gaps:

- **No severity distinction** — there is no way to treat a rule as a warning vs. an error
- **Two rules are hardcoded ON** — `unclosed-code-block` and `empty-alt-text` cannot be disabled via config

---

## Proposed Design

Adopt a **rule-map configuration** similar to markdownlint, replacing per-rule boolean flags with a unified `rules` object.

### New `.gomarklint.json` schema

```json
{
  "default": true,
  "rules": {
    "heading-level": { "enabled": true, "severity": "error", "minLevel": 2 },
    "duplicate-heading": true,
    "no-multiple-blank-lines": true,
    "fenced-code-language": "warning",
    "single-h1": false
  },
  "ignore": ["node_modules/**", "vendor/**"],
  "output": "text"
}
```

**Rule value forms (shorthand supported):**

| Value | Meaning |
|---|---|
| `true` | enabled, severity = `"error"` |
| `false` | disabled |
| `"warning"` | enabled, severity = `"warning"` |
| `{ "enabled": true, "severity": "warning", ...options }` | full form |

**`default` key:** when set to `false`, all rules are disabled unless explicitly listed — useful for strict opt-in configurations.

### New `Config` struct (Go)

```go
type RuleSeverity string

const (
    SeverityError   RuleSeverity = "error"
    SeverityWarning RuleSeverity = "warning"
    SeverityOff     RuleSeverity = "off"
)

type RuleConfig struct {
    Severity RuleSeverity
    Options  map[string]interface{}
}

type Config struct {
    Default bool                   `json:"default"`
    Rules   map[string]*RuleConfig `json:"rules"`
    Ignore  []string               `json:"ignore"`
    Output  string                 `json:"output"`
}
```

`RuleConfig` needs a custom `UnmarshalJSON` to handle all three shorthand forms.

### CLI flags

Rule-level control belongs in the config file, not on the command line. Per-rule CLI flags (e.g. `--enable-xxx-check`) add complexity without practical benefit — users reaching for that level of control will simply edit `.gomarklint.json`.

The CLI should expose only flags that are genuinely useful at invocation time:

| Flag | Description |
|---|---|
| `--config <path>` | Path to config file (useful for CI or multi-project setups) |
| `--output <format>` | Output format: `text` or `json` |
| `--severity <level>` | Minimum severity to report: `warning` or `error` (default: `warning`) |

All existing per-rule flags (`--enable-heading-level-check`, `--min-heading`, etc.) are removed.

---

## Breaking Changes → Major Version Bump to v2

This refactor introduces breaking changes to the configuration format:

1. All existing `enableXxxCheck` boolean fields are removed
2. `minHeadingLevel` moves into `rules["heading-level"]` options
3. All per-rule CLI flags are removed; only `--config`, `--output`, and `--severity` remain
4. Go module path changes from `github.com/shinagawa-web/gomarklint` to `github.com/shinagawa-web/gomarklint/v2`

---

## Implementation Steps

- [ ] Define `RuleConfig` type with custom JSON unmarshaler (handles `bool`, `string`, and object forms)
- [ ] Update `Config` struct — remove all `EnableXxxCheck` fields
- [ ] Update `load.go` / `merge.go` for new schema
- [ ] Update `linter.go` — drive rule execution from `rules` map
- [ ] Simplify CLI flags in `cmd/root.go` — keep only `--config`, `--output`, `--severity`
- [ ] Update `gomarklint init` to generate new schema
- [ ] Bump Go module path to `v2`
- [ ] Update all tests and benchmarks
- [ ] Add `docs/content/docs/migration-v2.md` to the GitHub Pages site — field-by-field mapping from v1 keys to v2 `rules` entries, CLI flag removal table, and a quick checklist

---

## Relation to Other Issues

This is a **prerequisite for #76** — all new rules added in that epic should use the new rule-map system from day one.
